### DISCUSSION TITLE
Memory Crash Issue - When will a solution be released?

### DISCUSSION THREAD

**Discussion by stelzk2** [cypress-io/cypress#30720] [Questions and Help]
Hello, I am wondering what the status of the memory leak/crash issue that Cypress experiences is. https://github.com/cypress-io/cypress/issues/27415

- What is the estimated date for a solution? 
- Has anyone found workarounds for this other than increasing the "--max-old-space-size=4096" launch parameter that will actually fix the memory leak? 

We need to be able to run reliability testing on our application, in which we run a single test for longer than 40 minutes, preferably for hours on end. We will have to look at other testing frameworks if this problem is not solved.

**Comment by portyu9** [comment]
Hi @stelzk2 

There is now a concrete update to the issue you linked.

**#27415 has been closed**, and Cypress **15.18.0** shipped a renderer-memory fix specifically addressing it.

That fix targets applications that repeatedly throw the same uncaught exception—for example a `ResizeObserver` exception firing every animation frame—which previously created repeated Command Log entries/DOM snapshots until renderer memory was exhausted.

So the first recommendation today is to upgrade to **Cypress 15.18.0 or later**, preferably the current supported Cypress release.

Cypress 16 also replaced:

```ts
experimentalMemoryManagement
```

with:

```ts
manageBrowserMemory
```

and browser memory management now defaults to:

```ts
manageBrowserMemory: true
```

for Chromium-based browsers.

However, there is an important limitation for your particular requirement: you want **one `it()` to run for 40 minutes or even hours**.

Current `manageBrowserMemory` performs its forced cleanup **between tests**. It therefore cannot completely solve memory that continually accumulates inside one enormous test.

Likewise:

```ts
numTestsKeptInMemory
```

controls snapshots/command data retained from completed tests; it does not make a single test's continuously growing command history disappear.

If possible, the strongest workaround is to split the reliability scenario into smaller test segments. For a dedicated endurance suite where retaining the browser page between tests is required, Cypress E2E also supports suite-level:

```js
describe('reliability test', {
  testIsolation: false,
}, () => {
  // sequential reliability segments
})
```

That does make the tests intentionally order-dependent, so I would reserve it for this special reliability/endurance suite rather than normal E2E tests.

Also prefer Chrome/Chromium over Electron for this use case. Electron is now deprecated, and there are still open reports involving very large in-test command queues and long-running Electron renderer memory usage.

One other distinction: if your `--max-old-space-size=4096` is being applied through **Node's** heap settings, that does not increase the Chromium renderer's heap. A Chromium `--js-flags` setting is a different mechanism.

So the situation today is:

- **#27415 itself was addressed in 15.18.0**
- Cypress 16 manages Chromium memory automatically between tests
- a single hours-long test can still be a special case because Cypress retains state/command information during that test
- splitting endurance work into test segments is still the safest design where possible

Please mark this answer as accepted if it helped, thank you.
